internal/ctxutil/ctx_test.go

```diff
@@ -0,0 +1,96 @@
+package ctxutil_test
+
+import (
+	"context"
+	"github.com/TykTechnologies/tyk/internal/ctxutil"
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func createReq(tb testing.TB) *http.Request {
+	tb.Helper()
+	req, err := http.NewRequest(http.MethodGet, "http://example.com", nil)
+	assert.NoError(tb, err)
+	return req
+}
+
+type mockUpstreamAuthProvider struct{}
```

Comment:
Generate a proper mock from the interface

Reply:
This shouldn't leave here. If it's an accessible mock object, it shouldn't be declared in tests, should be importable by tests code;

Similar RPC mock exists under model, also the interface for the filler could be in model/ too (same location).